src/CMakeLists.txt

```diff
@@ -25,6 +26,8 @@ else()
   set(REDIS_MODULE_LDFLAGS -shared)
 endif()
 
+add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/etcd)
```

Comment:
I think we should add an option to toggle the building of `<credis>/etcd`.  See: https://github.com/ray-project/ray/blob/master/CMakeLists.txt#L30

The reason is that building it takes too much time, so we should default the flag to off.

Reply:
What kind of build time increase are you seeing? I would prefer not to do this if we can fix the build time instead. Reasoning: it will require me to make significant changes to member.cc without any benefit other than shorter build times which should be solved with caching anyway.

Here is what I would need to do to make this change work:

1. Figure out how to attach instance methods to an aeEventLoop - because of this:
```
class Foo {
# instancefunc returns type Foo::*
void* instancefunc()
}

# globalfunc returns type void*
void* globalfunc
```
aeEventLoop callbacks have void* return type so it is difficult to coerce a non-global function into the right type without doing things that make me worry about memory safety, like stealing a raw function pointer from the instance by casting it to int and back. Hopefully you already figured this out working on the refresh_head/tail logic 😁 

2. Move all heartbeat-related code behind an abstraction that doesn't rely on any symbols from src/etcd.

3. Fix any resulting bugs.